### PR TITLE
[dualtor]: Make `force_active_tor` a fixture

### PR DESCRIPTION
**Pull request by theasianpianist**
* Enables automatic teardown/cleanup

Signed-off-by: Lawrence Lee <lawlee@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This util runs `config mux mode active` on a DUT, which may leave it in an unexpected state when starting the next test.

#### How did you do it?
Use the 'fixture yields function' pattern to enable automatic cleanup (running `config mux mode auto` for the affected ports during teardown)

### CODE REVIEW

File: tests/common/dualtor/dual_tor_utils.py [hunk 1]
```diff
@@ -185,19 +185,31 @@ def update_mux_configs_and_config_reload(dut, state):
...
+        else:
+            cmds = []
+            for i in intf:
+                forced_intfs.append((dut, i))
+                cmds.append("config muxcable mode active {}; true".format(i))
+        dut.shell_cmds(cmds=cmds, continue_on_fail=True)
+
+    yield _force_active_tor
+
+    if bool(forced_intfs):
```
Comment (vaibhavhd) [inline tests/common/dualtor/dual_tor_utils.py:209]: This bool check may not be needed as the below loop would be iterated only if `forced_intfs` list contains something.

File: tests/common/dualtor/dual_tor_utils.py [hunk 2]
```diff
@@ -185,19 +185,31 @@ def update_mux_configs_and_config_reload(dut, state):
...
     """
-    if type(intf) == str:
-        cmds = ["config muxcable mode active {}".format(intf)]
-    else:
-        cmds = []
-        for i in intf:
-            cmds.append("config muxcable mode active {}".format(i))
-    dut.shell_cmds(cmds=cmds)
+    forced_intfs = []
+    def _force_active_tor(dut, intf):
```
Comment (vaibhavhd) [inline tests/common/dualtor/dual_tor_utils.py:196]: Minor - to me it seems counterintuitive to define an internal function (leading underscore), and then yield it too for external use :)

**Review by vaibhavhd — APPROVED** [review]
Minor comments, LGTM otherwise.

File: tests/common/dualtor/dual_tor_utils.py (hunk 2 above)
Comment (theasianpianist) [reply tests/common/dualtor/dual_tor_utils.py:196]: True, though I think we already do it elsewhere so I just followed that convention. I'll change it

File: tests/common/dualtor/dual_tor_utils.py (hunk 1 above)
Comment (theasianpianist) [reply tests/common/dualtor/dual_tor_utils.py:209]: True, will remove

**Review by bingwang-ms — APPROVED** [review]

**Pull request merged**